### PR TITLE
Fixes #58: Check if a F-Droid suggestedVersionCode exists

### LINKED ISSUE #58: Automatic WireGuard download on Android fails due to F-Droid API quirk
Currently, the automatic WireGuard download on Android fails. The downloaded APK file is actually an HTML file of a 404 page.

Here's what `fdroidMeta` from the F-Droid API looks like:

```json
{
    "packageName": "com.wireguard.android",
    "suggestedVersionCode": 498,
    "packages": [
        { "versionName": "1.0.20230330", "versionCode": 496 },
        { "versionName": "1.0.20220516", "versionCode": 492 },
        { "versionName": "1.0.20211029", "versionCode": 491 }
    ]
}
```

From that, we build the following `apkUrl` based on the `suggestedVersionCode`: `https://f-droid.org/repo/com.wireguard.android_498.apk`

I had assumed that you were supposed to download the version according to the `suggestedVersionCode` (silly me :D), but it looks like that doesn't necessarily have to exist…

The F-Droid website currently list `496` as the newest available version, and the download for that works. So, I guess it's possible that `suggestedVersionCode` is incremented before that version is actually uploaded.

I guess we have to change our algorithm to:

1. Check whether `suggestedVersionCode` is present in `packages`. If yes, download that.
2. If not, download the highest `versionCode` from `packages`.

### PR DESCRIPTION
**Pull request by zner0L**

### CODE REVIEW

File: src/android.ts [hunk 1]
```diff
@@ -310,10 +310,21 @@ export const androidApi = <RunTarget extends SupportedRunTarget<'android'>>(
...
+                        .then((fdroidMeta) => {
+                            if (
+                                fdroidMeta.packages.some(
+                                    (p: { versionCode: number }) => p.versionCode === fdroidMeta.suggestedVersionCode
+                                )
+                            )
+                                return fdroidMeta.suggestedVersionCode;
+
+                            return fdroidMeta.packages.sort((a: { versionCode: number }, b: { versionCode: number }) =>
+                                a.versionCode === b.versionCode ? 0 : a.versionCode > b.versionCode ? 1 : -1
```
Comment (baltpeter) [inline src/android.ts:324]: You can just subtract the numbers, you don't need to return exactly `1` or `-1`.

Comment (baltpeter) [reply src/android.ts:324]: Also, this chooses the oldest version, which we don't want.

File: src/android.ts [hunk 2]
```diff
@@ -310,10 +310,21 @@ export const androidApi = <RunTarget extends SupportedRunTarget<'android'>>(
...
+                            )[0].versionCode;
+                        });
```
Comment (baltpeter) [inline src/android.ts:326]:
```suggestion
                            return fdroidMeta.packages.sort(
                                (a: { versionCode: number }, b: { versionCode: number }) =>
                                    b.versionCode - a.versionCode
                            )[0].versionCode;

```

**Review by baltpeter — CHANGES_REQUESTED** [review]

File: src/android.ts (hunk 1 above)
Comment (zner0L) [reply src/android.ts:324]: Ah, true. It was kinda late when I did that.

File: src/android.ts (hunk 2 above)
Comment (zner0L) [reply src/android.ts:326]: This is also the wrong way around, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

Comment (baltpeter) [reply src/android.ts]: https://pastebin.com/raw/gJGddmEF

**Review by baltpeter — APPROVED** [review]
Thank you!

**Pull request merged**